### PR TITLE
fix(hits): update lifecycle state

### PR DESCRIPTION
**Pull request by francoischalifour**
## Description

This fixes the lifecycle of `connectHits`.

## Changes

- `getConfiguration` returns a state computed with `SearchParameters`

### CODE REVIEW

File: src/connectors/hits/connectHits.js [hunk 1]
```diff
@@ -60,8 +60,14 @@ export default function connectHits(renderFn, unmountFn = noop) {
...
 
-      getConfiguration() {
-        return escapeHTML ? TAG_PLACEHOLDER : undefined;
+      getConfiguration(state) {
+        if (!escapeHTML) {
+          return state;
+        }
+
+        return state.setQueryParameters({
+          ...TAG_PLACEHOLDER,
```
Comment (Haroenv) [inline src/connectors/hits/connectHits.js]: feel free to ignore this; but `setQueryParameters(TAG_PLACEHOLDER)` is an option too

File: src/connectors/hits/connectHits.js [hunk 2]
```diff
@@ -60,8 +60,21 @@ export default function connectHits(renderFn, unmountFn = noop) {
...
-        return escapeHTML ? TAG_PLACEHOLDER : undefined;
+      getConfiguration(state) {
+        if (!escapeHTML) {
+          return state;
+        }
+
+        return state.setQueryParameters({
+          highlightPreTag:
+            state.highlightPreTag === undefined
+              ? TAG_PLACEHOLDER.highlightPreTag
```
Comment (Haroenv) [inline src/connectors/hits/connectHits.js]: I feel like we should be always overriding it, but I'm not fully sure

Comment (francoischalifour) [reply src/connectors/hits/connectHits.js]: I removed it.

File: src/connectors/hits/connectHits.js (hunk 1 above)
Comment (samouss) [reply src/connectors/hits/connectHits.js]: Agree with @Haroenv the spread is useless in that case.

**Pull request merged**